src/oss-tests/ProjectManagerTests/NuGetProjectManagerTests.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+
+namespace Microsoft.CST.OpenSource.Tests.ProjectManagerTests
+{
+    using Model;
+    using Moq;
+    using oss;
+    using PackageManagers;
+    using PackageUrl;
+    using RichardSzalay.MockHttp;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class NuGetProjectManagerTests
+    {
+        private readonly IDictionary<string, string> _packages = new Dictionary<string, string>()
+        {
+            { "https://api.nuget.org/v3/registration5-gz-semver2/razorengine/index.json", Resources.razorengine_json },
+            { "https://api.nuget.org/v3/catalog0/data/2022.03.11.23.17.27/razorengine.4.2.3-beta1.json", Resources.razorengine_4_2_3_beta1_json },
+        };
+
+        private NuGetProjectManager ProjectManager;
+        
+        public NuGetProjectManagerTests()
+        {
+            Mock<IHttpClientFactory> mockFactory = new();
+            
+            MockHttpMessageHandler mockHttp = new();
+
+            foreach ((string url, string json) in _packages)
+            {
+                MockHttpFetchResponse(HttpStatusCode.OK, url, json, mockHttp);
+            }
+ 
+            mockFactory.Setup(_ => _.CreateClient(It.IsAny<string>())).Returns(mockHttp.ToHttpClient());
+            
+            ProjectManager = new NuGetProjectManager(mockFactory.Object, ".");
+        }
+
+        [DataTestMethod]
+        [DataRow("pkg:nuget/razorengine@4.2.3-beta1", "RazorEngine - A Templating Engine based on the Razor parser.")] // Normal package
+        public async Task MetadataSucceeds(string purlString, string? description = null)
+        {
+            PackageURL purl = new(purlString);
+            PackageMetadata metadata = await ProjectManager.GetPackageMetadata(purl, useCache: false);
```

Comment:
Hmm - capitalizing the field name like this makes me think that this is a static method call, when it is really not a static method call. Have you considered using camelCase for field names instead?

```suggestion
            PackageMetadata metadata = await projectManager.GetPackageMetadata(purl, useCache: false);
```

Reply:
Fixed!